expr/functions/aliasByMetric/function.go

```diff
@@ -30,7 +31,7 @@ func New(configFile string) []interfaces.FunctionMetadata {
 
 func (f *aliasByMetric) Do(ctx context.Context, e parser.Expr, from, until int64, values map[parser.MetricRequest][]*types.MetricData) ([]*types.MetricData, error) {
 	return helper.ForEachSeriesDo1(ctx, e, from, until, values, func(a *types.MetricData) *types.MetricData {
-		metric := a.Tags["name"]
+		metric := metric.ExtractMetric(a.Name)
```

Comment:
Really needed ? It's more cost then simple use tag name. which already contain all, that we need (because all functions must keep tags consistent).

Reply:
We noticed the previous version was failing for cases where the nested result was named after the function that processed it. For example:

`aliasByMetric(ifft(foo.bar))`'


In that case, the argument of `aliasByMetric` is a series with name `ifft(foo.bar)`, which is not a metric name and the `strings.Split(metric, ".")` gives wrong results: `ifft(foo`, `bar)`. So we fixed it this was, by correctly extracting the metric name. Let us know if there's a better way.

Reply:
This is related to the other discussion we had. I've created this issue to address it later: https://github.com/go-graphite/carbonapi/issues/691